### PR TITLE
Recipe-App

### PR DESCRIPTION
**Pull request by jerryowusu**
## Description
The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe. Also, since sharing recipes is an important part of cooking the app should allow you to make them public so anyone can access them.

## General requirements
 

- [x] There are no linter errors.
 - [x] Followed the Github flow.
 - [x] The documentation is professional.

## Ruby requirements
- [x] Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

## Built With
Ruby
Rails
Postgresql
Devise
Bootstrap CSS
Rspec
Capybara

## Milestones 
- [x]  Milestone 1: A login page.
- [x] Milestone 2: A registration page.
- [x] Milestone 3: A food list (with all CRUD implementation, except for 'update').
- [x] Milestone 4: A recipes list (with all CRUD implementation, except for 'update').
 - [x] Milestone 6: A list of all public recipes from other users with their names and total prices.
 
### Additional information
Student success has agreed for me to run solo due to collaboration issues with my partner
[This](https://mail.google.com/mail/u/0/#inbox/FMfcgzGpGdhFQkVPGrSvHHCcjwMhGNGf) is the link to the authorization mail.

### Hope you enjoy yourself whiles reviewing :grinning:

### CODE REVIEW

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @jerryowusu @direakanbi, This is @sja-thedude your code reviewer.

While you made a great effort in this project, I cannot proceed to review your code.

You have submitted a project, that doesn't have commits from both members in your group, also linter checks are not present. Kindly report to student success if your partner is not responding and after receiving the message to submit alone, attach the screenshot in this pull request.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/models/food.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Food < ApplicationRecord
+  belongs_to :user
+  has_many :recipe_foods
+  has_many :recipes, through: :recipe_foods
+
+  # has_many :inventory_foods
+  # has_many :inventories, through: :inventory_foods
```
Comment (Davidosky007) [inline app/models/food.rb]: * Kindly remove all commented code from your codebase as it makes your code look messy and unprofesional 🙏

File: app/models/recipe_food.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class RecipeFood < ApplicationRecord
+  belongs_to :recipe
+  belongs_to :food
+
+  validates :quantity, numericality: { greater_than_or_equal_to: 1 }
+  # validates :food_id, presence: true
+  # validates :recipe_id, presence: true
```
Comment (Davidosky007) [inline app/models/recipe_food.rb]: * Kindly remove all commented code as mentioned above ⬆️

File: app/views/users/_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        </li>
+
+      
+         <%= render 'shared/sign_in_out' %>
+        
+       
+      </ul>
+      <%# <form class="d-flex" role="search">
+        <input class="form-control me-2" type="search" placeholder="Search" aria-label="Search">
+        <button class="btn btn-outline-success" type="submit">Search</button>
```
Comment (Davidosky007) [inline app/views/users/_header.html.erb:32]: * Kindly remove commented codes

File: app/controllers/publics_controller.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    end
+  end
+
+  # PATCH/PUT /publics/1 or /publics/1.json
+  def update
+    respond_to do |format|
+      if @public.update(public_params)
+        format.html { redirect_to public_url(@public), notice: 'Public was successfully updated.' }
+        format.json { render :show, status: :ok, location: @public }
+      else
```
Comment (Davidosky007) [inline app/controllers/publics_controller.rb:35]: * Kindly make some changes on your public recipe page to look like the wireframe given 
## YOURS 
image
##DESIGN 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @jerryowusu 👋, 
## Highlights ⭐ 
* No linter check errors 👍 
* Good readme file 👍 
* Professional PR 💯 
Good job so far!
## There are some issues that you still need to work on to go to the next project but you are ## almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
## [OPTIONAL] 
* I marked this test errors as optional because i can see in the project requirements it wasnt stated clearly to add units and integration tests, but i strongly suggest you fix these errors or update your readme files with the specific steps and commands to run them succesfully. below is a screenshot of the error i get .
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/food.rb (hunk 1 above)
Comment (jerryowusu) [reply app/models/food.rb]: Resolved. Thank you for drawing my attention.

File: app/models/recipe_food.rb (hunk 2 above)
Comment (jerryowusu) [reply app/models/recipe_food.rb]: Resolved. Thank you for drawing my attention.

File: app/views/users/_header.html.erb (hunk 3 above)
Comment (jerryowusu) [reply app/views/users/_header.html.erb:32]: Resolved. Thank you for drawing my attention.

File: app/controllers/publics_controller.rb (hunk 4 above)
Comment (jerryowusu) [reply app/controllers/publics_controller.rb:35]: Resolved. Thanks

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @jerryowusu ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by jerryowusu** [comment]
Thanks a lot, @VICTOR-CHUKWU and @Davidosky007   for reviewing my project and providing the necessary guidelines

**Pull request merged**